### PR TITLE
The Evolution of Bar RP

### PR DESCRIPTION
**Pull request by thgvr**
## About The Pull Request
Ports Saunas & Towels from Skyrat. Some edits to the code to fit here, and to make it work a little more sane. Ports some improvements to radial menus from TG.
image

## Changelog

:cl:
add: Saunas, craftable with wooden planks. Uses wood for fuel, and requires water splashed on the sauna.
add: Towels. You can use them in-hand to change it from waist, chest, or head.
/:cl:

### CODE REVIEW

**Comment by Martinpachu** [comment]
YEAAA YEAAH

**Comment by Norletic** [comment]
love dieing of liver issues while drinking pure ethanol in a sauna

**Comment by Martinpachu** [comment]
the ''on'' and ''off'' messages are reversed. turning it on says it turns off etc

**Comment by thgvr** [comment]
I already fixed that

File: code/game/objects/structures/sauna.dm [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		switch(water_amount) // Stops infinite vapor generation from shady TOUCH methods like showers.
+			if(0)
+				return
+			if(1)
+				used_amount = 1
+			if(2 to 199)
+				used_amount = water_amount * 0.1
+			else
+				used_amount = 20
+		water_amount -= used_amount
```
Comment (MarkSuckerberg) [inline code/game/objects/structures/sauna.dm]:
```suggestion
		var/used_amount = min(water_amount / 10, 20)
		water_amount -= used_amount
```

**Review by MarkSuckerberg — DISMISSED** [review]

**Pull request merged**